src/unix/epoll.rs

```diff
@@ -0,0 +1,429 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Wrapper over `libc::epoll` API.
+
+use std::io;
+use std::ops::Deref;
+use std::os::raw::c_int;
+use std::os::unix::io::{AsRawFd, RawFd};
+
+use libc::{
+    epoll_create1, epoll_ctl, epoll_event, epoll_wait, EPOLLERR, EPOLLET, EPOLLEXCLUSIVE, EPOLLHUP,
+    EPOLLIN, EPOLLONESHOT, EPOLLOUT, EPOLLPRI, EPOLLRDHUP, EPOLLWAKEUP, EPOLL_CLOEXEC,
+    EPOLL_CTL_ADD, EPOLL_CTL_DEL, EPOLL_CTL_MOD,
+};
+
+/// Wrapper to interpret syscall exit codes and provide a rustacean `io::Result`
+pub struct SyscallReturnCode(pub c_int);
```

Comment:
This shouldn't be in the `epoll` mod but separate. It operates with any syscall.

Reply:
I moved it to a new file, `syscall.rs`, as I didn't find a suitable place in the existing files. What do you think?